### PR TITLE
rt-tests: init at 2.3

### PR DESCRIPTION
**Pull request by poelzi**
###### Description of changes
Add realtime tests suite

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/rt-tests/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, lib, fetchurl, pkgs }:
```
Comment (muscaln) [inline pkgs/os-specific/linux/rt-tests/default.nix:1]:
```suggestion
{ stdenv
, lib
, makeWrapper
, fetchurl
, numactl
, python3
}:
```

File: pkgs/os-specific/linux/rt-tests/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+stdenv.mkDerivation rec {
+  pname = "rt-tests";
+  version = "2.3";
+
+  src = fetchurl {
+    url = "https://git.kernel.org/pub/scm/utils/rt-tests/rt-tests.git/snapshot/${pname}-${version}.tar.gz";
+    sha256 = "Q+rNdpRdsmW2gcsrfwg12EzpvO6qlEP/Mb/OWQMNmr8=";
+  };
+
+  buildInputs = with pkgs;[ numactl python3 makeWrapper];
```
Comment (muscaln) [inline pkgs/os-specific/linux/rt-tests/default.nix:12]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
  buildInputs = [ numactl python3 ];
```

**Review by muscaln — CHANGES_REQUESTED** [review]

**Comment by peng1999** [comment]
Hi, @poelzi 
This package will install two python modules into `$(out)/lib/python/`. But currently it is not work well with `python3.withPackages` and we cannot run the modules by running `python -m hwlatdetect` without manually setting `PYTHONPATH` to an obscure `/nix/store` path. Is it possible to fix this?

**Comment by poelzi** [comment]
@peng1999 the installer is quite strange and wants to place the python script inside the site-packages. I did not think anybody will run this as a python module but only use the bin symlink installed. I will look how to fix this if running this as a python module is that 
important.

**Comment by poelzi** [comment]
@peng1999 better now ?

**Comment by peng1999** [comment]
@poelzi Thanks for your improvement.

File: pkgs/os-specific/linux/rt-tests/default.nix (hunk 1 above)
Comment (Emantor) [reply pkgs/os-specific/linux/rt-tests/default.nix:1]: Why has this been marked as resolved?

**Comment by muscaln** [comment]
@poelzi It looks like you just did a wrong force-push. Could you re-apply the suggested changes?

File: pkgs/os-specific/linux/rt-tests/default.nix (hunk 1 above)
Comment (poelzi) [reply pkgs/os-specific/linux/rt-tests/default.nix]: because I have messed up and overwrote this change. fixed now

**Comment by poelzi** [comment]
@muscaln damn it, i messed up :-) Should be good now

File: pkgs/os-specific/linux/rt-tests/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchurl {
+    url = "https://git.kernel.org/pub/scm/utils/rt-tests/rt-tests.git/snapshot/${pname}-${version}.tar.gz";
+    sha256 = "Q+rNdpRdsmW2gcsrfwg12EzpvO6qlEP/Mb/OWQMNmr8=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ numactl python3 ];
+
+  makeFlags = [ "prefix=$(out)" "DESTDIR=" "PYLIB=$(out)/${pkgs.python3.sitePackages}" ];
```
Comment (muscaln) [inline pkgs/os-specific/linux/rt-tests/default.nix:21]:
```suggestion
  makeFlags = [ "prefix=$(out)" "DESTDIR=" "PYLIB=$(out)/${python3.sitePackages}" ];
```

**Review by muscaln — CHANGES_REQUESTED** [review]

**Pull request merged**